From a620077e13f16a6ddbdede774fdc2915fbfc364d Mon Sep 17 00:00:00 2001
From: Marc Lehmann <schmorp@schmorp.de>
Date: Thu, 30 Jun 2016 14:07:43 +0000
Subject: [PATCH 11/31] *** empty log message ***

---
 src/command.C | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/command.C b/src/command.C
index dca5e51e5276..70ff277c9d51 100644
--- a/src/command.C
+++ b/src/command.C
@@ -3384,20 +3384,20 @@ rxvt_term::map_rgb24_color (unsigned int r, unsigned int g, unsigned int b)
 
   for (int n = 0; n < ecb_array_length (dxyz); ++n)
     {
-      int i = idx_r + dxyz[n][0];
-      int j = idx_r + dxyz[n][1];
-      int k = idx_r + dxyz[n][2];
+      int r = idx_r + dxyz[n][0];
+      int g = idx_r + dxyz[n][1];
+      int b = idx_r + dxyz[n][2];
 
-      if (!IN_RANGE_EXC (i, 0, Red_levels))
+      if (!IN_RANGE_EXC (r, 0, Red_levels))
         continue;
 
-      if (!IN_RANGE_EXC (j, 0, Green_levels))
+      if (!IN_RANGE_EXC (g, 0, Green_levels))
         continue;
 
-      if (!IN_RANGE_EXC (k, 0, Blue_levels))
+      if (!IN_RANGE_EXC (b, 0, Blue_levels))
         continue;
 
-      unsigned int index = colorcube_index (i, j, k);
+      unsigned int index = colorcube_index (r, g, b);
 
       // minor issue: could update index 0 few more times
       if ((rgb24_seqno[index] | rgb24_color[index]) == 0)
-- 
2.9.4

